tests/functional.mk

```diff
@@ -63,6 +85,30 @@ BGB_ADDITIONAL_GO_ENV := CGO_ENABLED=0
 
 include makelib/build_go_bin.mk
 
+$(FTST_ACE_MAIN_IMAGE): ACTOOL := $(ACTOOL)
+$(FTST_ACE_MAIN_IMAGE): FTST_ACE_MAIN_IMAGE_DIR := $(FTST_ACE_MAIN_IMAGE_DIR)
+$(FTST_ACE_MAIN_IMAGE): FTST_ACE_MAIN_IMAGE_ROOTFSDIR := $(FTST_ACE_MAIN_IMAGE_ROOTFSDIR)
+$(FTST_ACE_MAIN_IMAGE): $(FTST_ACE_MAIN_IMAGE_MANIFEST) $(ACEVALIDATOR_STAMP) | $(FTST_ACE_MAIN_IMAGE_ROOTFSDIR)
+	mkdir -p $(FTST_ACE_MAIN_IMAGE_ROOTFSDIR)/opt/acvalidator
```

Comment:
You mean generalization of the rule generating main and sidekick images (looks like they are the same)? If so, something like this (witness the power of quadruple dollar sign!):
```make
# 1 - image
# 2 - aci directory
# 3 - ace validator
define ftst-generate-ace-image
$1: ACTOOL := $$(ACTOOL)
$1: $2/manifest <ACE VALIDATOR IN $1> | $2/rootfs
	mkdir -p $2/opt/acvalidator
	# Set a consistent timestamp so we get a consistent hash
	# TODO(jonboulle): make this cleaner..
	for path in "$2/rootfs" "$2/rootfs/ace-validator"; do \
		touch -a -m -d 1970-01-01T00:00:00Z "$$$${path}"; \
	done
	"$$(ACTOOL)" build --overwrite "$2" "$@"

CREATE_DIRS += $2 $2/rootfs
INSTALL_FILES += $3:$2/rootfs/ace-validator (or something):-
endef
```
and then:
```make
$(eval $(call ftst-generate-ace-image,<PATH TO ACI IMAGE WE WANT TO GENERATE>,<PATH TO ACI DIRECTORY, NOT ROOTFS!>,<PATH TO BUILT ACE VALIDATOR>))
```